controllers/scenario/scenario_adapter_test.go

```diff
@@ -144,6 +147,35 @@ var _ = Describe("Scenario Adapter", Ordered, func() {
 		Expect(reflect.TypeOf(NewAdapter(hasApp, invalidScenario, ctrl.Log, k8sClient, ctx))).To(Equal(reflect.TypeOf(&Adapter{})))
 	})
 
+	It("EnsureCreatedScenarioIsValid without app", func() {
+		a := NewAdapter(nil, integrationTestScenario, ctrl.Log, k8sClient, ctx)
+
+		Eventually(func() bool {
+			result, err := a.EnsureCreatedScenarioIsValid()
+			return !result.CancelRequest && err == nil
+		}, time.Second*10).Should(BeTrue())
+
+	})
+
+	When("environment is in a different namespace than scenario", func() {
+		BeforeEach(func() {
+			envNamespace = "separatenamespace"
+		})
+
+		AfterEach(func() {
+			envNamespace = DefaultNamespace
+		})
+
+		It("ensure the scenario status is invalid", func() {
+			Eventually(func() bool {
+				result, err := adapter.EnsureCreatedScenarioIsValid()
+				return !result.CancelRequest && err == nil
+			}, time.Second*10).Should(BeTrue())
+			Expect(meta.IsStatusConditionFalse(integrationTestScenario.Status.Conditions, gitops.IntegrationTestScenarioValid)).To(BeTrue())
```

Comment:
I cannot figure out, why this test is failing, it's supposed to be Invalid

Reply:
I think it's because the [env](https://github.com/redhat-appstudio/integration-service/pull/95/files#diff-2ea0168978a368c1f4520a42d682ce29f826a4e0014a6dc573a4a2ef1964db4dR98) is created only in the `BeforeAll` function and is not modified or recreated before every test.

Reply:
I updated test to utilize JustBeforeEach, PTAL, it's little bit magic, but now it works :D

Reply:
Yes, this looks like the way to go